cson/cson.c

```diff
@@ -127,11 +127,13 @@ struct cson_value {
     int sub_type;
     int value_bytes;
     char *value_text;
+    char value_buf[48];
```

Comment:
This is good to have a optional buffer to use like this.
i was wondering if we could stick value_text at the end of the obj so for instance when you create an int we would do:
```
cson_value *cson_value_new_integer(cson_int_t v)
 {
     char buf[128];
     val->value_bytes = sprintf(buf, "%" PRId64, v);

     cson_value *val = calloc(1, sizeof(cson_value)  + val->value_bytes );
     val->sql_type = SQLITE_INTEGER;
     val->u.num = v;
     val->value_text = val + sizeof(cson_value); 
     strncpy(val->value_text, buf, val->value_bytes);
     return val;
 }
```

More work needed for correctness, you would need to not free() value_text, etc.

Reply:
We can revisit it in subsequent PR. Having fixed size buf lets us increase without new alloc. Imagine an array with 2 small ints. Initial allocation of array needs 3 bytes only `[`,`]`,`0`. On appending, we need to realloc.